toucan_connectors/redshift/redshift_database_connector.py

```diff
@@ -133,11 +148,16 @@ class RedshiftConnector(ToucanConnector, DiscoverableConnector):
 
     class Config:
         underscore_attrs_are_private = True
+        keep_untouched = (cached_property,)
 
         @staticmethod
         def schema_extra(schema: dict[str, Any]) -> None:
             schema['properties'] = {k: schema['properties'][k] for k in ORDERED_KEYS}
 
+    @cached_property
```

Comment:
:+1:

Reply:
one thought about it is when data get's large the cached property stay's forever in RAM.
```suggestion
    @cached_property_with_ttl(ttl=300) # five minutes ttl
```